### PR TITLE
Remove bonding curve

### PR DESCRIPTION
**Pull request by sstrenev**
In this PR:
 - Remove bonding curve for polymorph pricing - now it is set in the constructor and can be changed only by the DAO
 - Remove calculation for randomizeGenome price - now set in the constructor and changeable by the DAO
 - Add total supply so no more polymorphs could be minted if it's reached - settable in the constructor and changeable by the DAO
 - Add bulk buy option with up to 30 polymorphs in single buy

### CODE REVIEW

File: contracts/Polymorph.sol [hunk 1]
```diff
@@ -76,25 +80,54 @@ contract Polymorph is IPolymorph, ERC721PresetMinterPauserAutoId, BMath, Reentra
...
         emit TokenMinted(tokenId, _genes[tokenId]);
-        emit TokenMorphed(tokenId, 0, _genes[tokenId], price, PolymorphEventType.MINT);
+        emit TokenMorphed(tokenId, 0, _genes[tokenId], polymorphPrice, PolymorphEventType.MINT);
     }
 
-    function lastTokenId() public override view returns (uint256 tokenId) {
-        return _tokenIdTracker.current();
+    function bulkBuy(uint256 amount) public override payable nonReentrant {
+        require(msg.value >= polymorphPrice.mul(amount), "Not enough funds");
+        require(amount <= 30, "Cannot bulk buy more than 30");
```
Comment (Perseverance) [inline contracts/Polymorph.sol:88]: Lets make the 30 a parameter like the premint and make it changeable by the DAO

File: contracts/Polymorph.sol [hunk 2]
```diff
@@ -57,17 +60,18 @@ contract Polymorph is IPolymorph, ERC721PresetMinterPauserAutoId, BMath, Reentra
     }
 
     function mint() public override payable nonReentrant {
+        require(msg.value >= polymorphPrice, "Not enough funds");
```
Comment (Perseverance) [inline contracts/Polymorph.sol:63]: I dont think you need to check this, as both the `.call` and `.sub` will fail if less than the polymorph price

**Review by Perseverance — CHANGES_REQUESTED** [review]

File: contracts/Polymorph.sol (hunk 2 above)
Comment (strenev) [reply contracts/Polymorph.sol]: Done!

File: contracts/Polymorph.sol (hunk 1 above)
Comment (strenev) [reply contracts/Polymorph.sol]: Changed as well

**Pull request merged**